docs/upload_specifications.ADR.md

```diff
@@ -0,0 +1,170 @@
+# Upload Specifications Architecture Design Record
+
+This document specifies the design for handling upload specifications in the Staging Service (StS).
+An upload specification is an Excel, CSV, or TSV file that contains instructions for how
+to upload one or more files in the staging area to KBase as KBase data types.
+
+## Resources
+
+* [The original strategy document for this approach](https://docs.google.com/document/d/1ocmZVBlTzAh_cdZaWGRwIbAuH-mcPRZFdhcwRhAfzxM/edit)
+  * Readable for everyone that has access to the KBase Google Docs folder
+  * The document contains short descriptions of future projects not included in this work,
+    such as generating template files on the fly and rich templates.
+
+## Front end changes
+
+The design introduces a new StS data type, `upload_specification`. The FE's current 
+behavior is to display any data types in the file dropdown, but ignore files for which
+the selected data type is unknown, a bug. The FE will be updated to ignore unknown
+data types returned from the StS, allowing for phased, non-lockstep upgrades. This work is not
+including is this project, but will be in a future FE project.
+
+## Upload specification input files
+
+Input file formats may be Excel, CSV, or TSV. An example CSV file structure for GFF/FASTA uploads
```

Comment:
How will the csv/xls files associated with sampleset descriptions be handled by the staging service?

Reply:
No differently than any other upload file. Currently a sample upload looks like this:
![image](https://user-images.githubusercontent.com/1686141/127391881-705586e4-1a0b-46cb-b06e-3210658044f4.png)

When this is implemented, there will be another entry in the suggested types list, `Upload Specification`. The user would just pick the appropriate type.

Does that answer the question?

Reply:
Hmm, this is a place where an AI type picker could be pretty helpful - it should be reasonably easy to distinguish between samples, media, FBA, and upload specs.